crates/astria-core/Cargo.toml

```diff
@@ -55,6 +55,10 @@ server = ["dep:tonic"]
 test-utils = []
 base64-serde = ["dep:base64-serde"]
 brotli = ["dep:brotli"]
+# When enabled, this adds constructors for some types that skip the normal constructor validity
+# checks. It supports the case where the inputs are already deemed valid, e.g. having read them from
+# local storage.
+unchecked-constructors = []
```

Comment:
I like this solution. Really forces you to opt into this.

I think in addition to this flag, we should make a note on all feature-gated methods that we don't provide stability guarantees for them.

Reply:
I tend to assume `doc(hidden)` implies this, but I guess better to be explicit.

Updated in [ccc3f9e](https://github.com/astriaorg/astria/pull/1492/commits/ccc3f9e62661d8c6f0d73a52a119ed18aa89cc60).